stock_valued_picking_report/models/stock_pack_operation.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Pedro M. Baeza - Tecnativa <pedro.baeza@tecnativa.com>
+# Copyright 2015 Antonio Espinosa - Tecnativa <antonio.espinosa@tecnativa.com>
+# Copyright 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models, api
+from openerp.addons.decimal_precision import decimal_precision as dp
+
+
+class StockPackOperation(models.Model):
+    _inherit = "stock.pack.operation"
+
+    sale_line = fields.Many2one(
+        comodel_name='sale.order.line',
+        compute='compute_sale_order_line_fields',
+        string="Related order line", readonly=True)
+    sale_price_unit = fields.Float(
+        compute='compute_sale_order_line_fields',
+        digits=dp.get_precision('Product Price'),
+        string="Sale price unit", readonly=True)
+    sale_discount = fields.Float(
+        compute='compute_sale_order_line_fields',
+        digits=dp.get_precision('Discount'),
+        string="Sale discount (%)", readonly=True)
+    sale_price_subtotal = fields.Float(
+        compute='compute_sale_order_line_fields',
+        digits=dp.get_precision('Account'),
+        string="Price subtotal", readonly=True)
+
+    @api.multi
+    @api.depends(
+        'sale_price_unit',
+        'sale_discount',
+        'product_qty',
+        'sale_line.order_id.currency_id',
+    )
+    def compute_sale_order_line_fields(self):
+        for operation in self:
+            sale_lines = self.mapped(
+                'linked_move_operation_ids.move_id.procurement_id.'
+                'sale_line_id')
+            if len(sale_lines) <= 1:
+                operation.sale_price_unit = sale_lines.price_unit
+                operation.sale_discount = sale_lines.discount
+            else:
+                sum_qty = 0.0
+                sum_price = 0.0
+                sum_discount = 0.0
+                for sale_line in sale_lines:
+                    sum_qty += sale_line.product_qty
+                    sum_price += sale_line.price_unit * sale_line.product_qty
+                    sum_discount += sale_line.discount * sale_line.product_qty
+                operation.sale_price_unit = sum_price / sum_qty
+                operation.sale_discount = sum_discount / sum_qty
+
+            operation.sale_line = sale_lines[:1]
+            subtotal = (operation.sale_price_unit * operation.product_qty *
+                        (1 - (operation.sale_discount or 0.0) / 100.0))
+            currency = sale_lines.mapped('currency_id')
```

Comment:
What would happen if there were more than 1 record in `currency`? Is this case possible and, if so, controlled?